Mathlib/Algebra/CharP/ExpChar.lean

```diff
@@ -244,46 +247,152 @@ theorem ExpChar.neg_one_pow_expChar_pow [Ring R] (q n : ℕ) [hR : ExpChar R q]
   · simp only [one_pow, pow_one]
   haveI := Fact.mk hprime; exact CharP.neg_one_pow_char_pow R q n
 
-section BigOperators
+section frobenius
+
+section CommSemiring
+
+variable [CommSemiring R] {S : Type*} [CommSemiring S] (f : R →* S) (g : R →+* S) (p n : ℕ)
+  [ExpChar R p] [ExpChar S p] (x y : R)
+
+/-- The frobenius map that sends x to x^p -/
+def frobenius : R →+* R where
+  __ := powMonoidHom p
+  map_zero' := zero_pow (expChar_pos R p)
+  map_add' := add_pow_expChar R
+#align frobenius frobenius
+
+/-- The iterated frobenius map sending x to x^p^n -/
+def iterateFrobenius : R →+* R where
+  __ := powMonoidHom (p ^ n)
+  map_zero' := zero_pow (expChar_pow_pos R p n)
+  map_add' := add_pow_expChar_pow R
+
+variable {R}
+
+theorem frobenius_def : frobenius R p x = x ^ p :=
+  rfl
+#align frobenius_def frobenius_def
+
+theorem iterateFrobenius_def : iterateFrobenius R p n x = x ^ p ^ n := rfl
+
+theorem iterate_frobenius (n : ℕ) : (frobenius R p)^[n] x = x ^ p ^ n := by
+  induction n with
+  | zero => simp
+  | succ n n_ih =>
+      rw [Function.iterate_succ', pow_succ', pow_mul, Function.comp_apply, frobenius_def, n_ih]
```

Comment:
Maybe you can add a result which states that `iterateFrobenius R p n` is equal to `(frobenius R p)^[n]` and this result is a corollary of it?

More generally, `iterateFrobenius R p (m * n)` is equal to `(iterateFrobenius R p m)^[n]`, similar to [Function.iterate_mul](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Logic/Function/Iterate.html#Function.iterate_mul).

Also, `iterateFrobenius R p 1` is equal to `frobenius R p` and `iterateFrobenius R p 0` is equal to `RingHom.id`. Maybe these two should be simp lemmas?

Also, `iterateFrobenius R p (m + n)` is equal to `(iterateFrobenius R p m).comp (iterateFrobenius R p n)`.

---

Perhaps all of the above should be corollaries of corresponding results for `RingHom.iterate` (to be defined).

Reply:
> Maybe you can add a result which states that `iterateFrobenius R p n` is equal to `(frobenius R p)^[n]` and this result is a corollary of it?

Good idea; that should be probably called `coe_iterateFrobenius`.

The other ideas are also good! Please feel free to add commits (and the same applies to #9311).